### PR TITLE
Fix integration test and bump sdk to 1.9.27

### PR DESCRIPTION
**Pull request by zhang-hua**
# Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] The code follows the code style and conventions of this project.
- [x] All new and existing tests passed.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java [hunk 1]
```diff
@@ -19,9 +19,9 @@
     private String id = null;
     private boolean enabled = false;
     private long c2DMessageCount = 0;
-    private DateTime lastActivity = null;
+    private Date lastActivity = null;
     private boolean connected = false;
-    private DateTime lastStatusUpdated = null;
+    private Date lastStatusUpdated = null;
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: does this mean that hour/minute/second are not available anymore? does this match the .NET API?

File: app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java [hunk 2]
```diff
@@ -178,12 +186,12 @@ public DeviceServiceModel toServiceModel() {
             this.eTag,
             this.id,
             this.c2DMessageCount,
-            this.lastActivity,
+            new DateTime(this.lastActivity),
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: we should always use `DateTimeOffset` class for consistency. If an external SDK needs the old DateTime, you can use DateTimeOffset.DateTime

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java [hunk 3]
```diff
@@ -92,4 +92,11 @@ public void toAzureModelTest() {
             com.microsoft.azure.sdk.iot.service.jobs.JobStatus.queued,
             JobStatus.toAzureJobStatus(JobStatus.queued));
     }
+
+    @Test(timeout = 100000, expected = IllegalArgumentException.class)
+    @Category({UnitTest.class})
+    public void toAzureModelWithNullValueTest() {
+        JobStatus.toAzureJobStatus(null);
```
Comment (dluc) [inline test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java]: could you clarify the expectation of this test? if "no exception" should occur, please add a comment

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java [hunk 4]
```diff
@@ -19,7 +19,7 @@ public void fromValueTest() {
     @Test(timeout = 100000, expected = IllegalArgumentException.class)
     @Category({UnitTest.class})
     public void fromValueFailureTest() {
-        Assert.assertEquals(JobType.unknown, JobType.from(10));
+        JobType.from(10);
```
Comment (dluc) [inline test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java]: what is this test testing? please add some comment

File: app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java (hunk 1 above)
Comment (zhang-hua) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: they are still available. there are only three milliseconds compare to .NET.  e.g. 2017-10-09T09:54:58.054Z.

File: app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java (hunk 2 above)
Comment (zhang-hua) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: ok.

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java (hunk 3 above)
Comment (zhang-hua) [reply test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java]: the expected exception is in the annotation: 
Test(timeout = 100000, expected = IllegalArgumentException.class)

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java (hunk 4 above)
Comment (zhang-hua) [reply test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java]: Expect IllegalArgumentException in the annoatation:
@Test(timeout = 100000, expected = IllegalArgumentException.class)

File: app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java (hunk 2 above)
Comment (zhang-hua) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: DateTimeOffset is defined in .NET framework.  But for Java,  [java.time.OffsetDateTime] (https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html) is defined from version 1.8.  I did not see good example in pcs java projects to use this class.

Comment (dluc) [reply app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/models/DeviceRegistryApiModel.java]: my bad, thought this was a .NET PR for a second :-)

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java (hunk 3 above)
Comment (dluc) [reply test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobStatusTest.java]: 👍

File: test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java (hunk 4 above)
Comment (dluc) [reply test/com/microsoft/azure/iotsolutions/iothubmanager/services/models/JobTypeTest.java]: 👍

**Comment by dluc** [comment]
I was about to merge but the commit msg is too long, I'll leave it with you @zhang-hua 
Thanks for improving the tests

**Comment by zhang-hua** [comment]
@dluc, I don't have write access to merge the PR.

**Comment by dluc** [comment]
@zhang-hua I wanted to merge but I couldn't polish the commit message. Commit messages like this https://github.com/Azure/iothub-manager-java/commit/bbdbbf697370e4853393b53bd21695dc103ab6ed are pretty hard to understand

**Pull request merged**